openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java

```diff
@@ -477,7 +478,27 @@ public static String getOpenApiFileName(String servicePath, String serviceName,
             // Replace rest of the path separators with underscore
             openAPIFileName = serviceName.replaceAll(SLASH, "_");
         }
-        return openAPIFileName + Constants.OPENAPI_SUFFIX + (isJson ? JSON_EXTENSION : YAML_EXTENSION);
+
+        return getNormalizeFileName(openAPIFileName) + Constants.OPENAPI_SUFFIX +
+                (isJson ? JSON_EXTENSION : YAML_EXTENSION);
+    }
+
+    /**
+     * Remove special characters from the given file name.
+     */
+    public static String getNormalizeFileName(String openAPIFileName) {
+
+        String[] splitNames = openAPIFileName.split("[^a-zA-Z0-9]");
+        StringBuilder normalizeFileName = new StringBuilder();
+        for (String name: splitNames) {
+            if (!name.isBlank()) {
+                normalizeFileName.append(name).append(UNDERSCORE);
+            }
+        }
+        if (normalizeFileName.toString().endsWith(UNDERSCORE)) {
+            return normalizeFileName.deleteCharAt(normalizeFileName.length() - 1).toString();
+        }
```

Comment:
You can replace the `StringBuilder` with a `StringJoiner` instance for the same purpose, but with less complexity

Reply:
Yeah , will use `StringJoiner` for the implementation.